keras/src/layers/layer.py

```diff
@@ -306,21 +306,21 @@ def __init__(
         self._losses_override = []
 
         self._call_signature = inspect.signature(self.call)
-        call_signature_parameters = [
+        self.call_signature_parameters = [
             p.name for p in self._call_signature.parameters.values()
         ]
-        self._call_has_training_arg = "training" in call_signature_parameters
-        self._call_has_mask_arg = "mask" in call_signature_parameters
+        self._call_has_training_arg = (
+            "training" in self.call_signature_parameters
+        )
+        self._call_has_mask_arg = "mask" in self.call_signature_parameters
 
         # 1. collect names that should be auto‑propagated
-        builtin_context_args = {"training"}
-        custom_context_args = set(getattr(self, "call_context_args", ()))
-        self._call_ctx_args = builtin_context_args | custom_context_args
+        self._call_context_args = {"training"}
```

Comment:
Wait, this needs to move to `__init__`. Otherwise it will overwrite your registrations.

Reply:
Never mind, this is `__init__`.